test/ecto/changeset_test.exs

```diff
@@ -1044,6 +1045,58 @@ defmodule Ecto.ChangesetTest do
     assert changeset.errors == [terms_of_service: {"must be abided", [validation: :acceptance]}]
   end
 
+  test "validate_unique_tentatively/3" do
+    defmodule FakeRepoWithDup do
```

Comment:
I think @josevalim had an idea to modify the test adapter. In particular the function here: https://github.com/elixir-ecto/ecto/blob/master/test/support/test_repo.exs#L43-L45

We could make it use something like:
```
Process.get(:test_repo_all_result) || {1, [[1]]}
```

And then in the test case, we could set the result with `Process.put(:test_repo, all_result, ...)`.